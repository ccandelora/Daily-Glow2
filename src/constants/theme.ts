export const COLORS = {
  // Primary colors for emotions (keeping these for emotional context)
  primary: {
    red: '#FF6B6B',     // Warm red for negative emotions
    green: '#00C853',   // Vibrant green for positive emotions
    blue: '#2979FF',    // Bright blue for calm emotions
    yellow: '#FFD600',  // Bright yellow for energetic emotions
  },
  
  // UI colors updated for dark theme with glowing accents
  ui: {
    background: 'rgba(28, 14, 45, 0.95)',  // Dark purple background
    card: 'rgba(38, 20, 60, 0.85)',        // Slightly lighter purple for cards
    text: '#FFFFFF',                        // White text for contrast
    textSecondary: 'rgba(255, 255, 255, 0.7)', // Semi-transparent white
    border: 'rgba(82, 67, 194, 0.3)',      // Subtle purple border
    disabled: 'rgba(82, 67, 194, 0.2)',    // Dimmed purple for disabled state
    accent: '#4169E1',                     // Royal blue for accents
    highlight: 'rgba(65, 105, 225, 0.15)', // Soft blue highlight
  },

  // Status colors with glow effects
  status: {
    success: '#00E676',
    error: '#FF5252',
    warning: '#FFD600',
    info: '#4169E1',
  },

  // Gradient colors for glow effects
  gradient: {
    start: 'rgba(65, 105, 225, 0.2)',  // Royal blue glow
    middle: 'rgba(147, 112, 219, 0.15)', // Purple glow
    end: 'rgba(28, 14, 45, 0.9)',     // Dark purple base
  },
};

export const FONTS = {
  sizes: {
    xs: 12,
    sm: 14,
    md: 16,
    lg: 22,
    xl: 28,
    xxl: 36,
    xxxl: 44,
  },
  weights: {
    regular: '400' as const,
    medium: '500' as const,
    semibold: '600' as const,
    bold: '700' as const,
  },
  // Adding font families for more sophistication
  families: {
    heading: 'System',
    body: 'System',
  }
};

export const SPACING = {
  xs: 4,
  sm: 8,
  md: 16,
  lg: 24,
  xl: 32,
  xxl: 40,
};

export const BORDER_RADIUS = {
  sm: 4,
  md: 8,
  lg: 16,
  xl: 24,
  circle: 999,
};

export const TIME_PERIODS = {
  MORNING: {
    label: 'Morning',
    range: { start: 5, end: 11 }, // 5:00 AM - 11:59 AM
    greeting: 'Good morning',
    icon: 'üåÖ'
  },
  AFTERNOON: {
    label: 'Afternoon',
    range: { start: 12, end: 16 }, // 12:00 PM - 4:59 PM
    greeting: 'Good afternoon',
    icon: '‚òÄÔ∏è'
  },
  EVENING: {
    label: 'Evening',
    range: { start: 17, end: 4 }, // 5:00 PM - 4:59 AM
    greeting: 'Good evening',
    icon: 'ÔøΩÔøΩ'
  }
} as const;

export default {
  COLORS,
  FONTS,
  SPACING,
  BORDER_RADIUS,
  TIME_PERIODS,
}; 