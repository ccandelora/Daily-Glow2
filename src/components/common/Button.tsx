import React from 'react';
import { TouchableOpacity, StyleSheet, ViewStyle, TextStyle, ActivityIndicator, StyleProp } from 'react-native';
import { Typography } from './Typography';
import theme from '@/constants/theme';

interface ButtonProps {
  title: string;
  onPress: () => void;
  variant?: 'primary' | 'secondary' | 'outline';
  size?: 'small' | 'medium' | 'large';
  disabled?: boolean;
  loading?: boolean;
  style?: ViewStyle;
  textStyle?: TextStyle;
}

export const Button: React.FC<ButtonProps> = ({
  title,
  onPress,
  variant = 'primary',
  size = 'medium',
  disabled = false,
  loading = false,
  style,
  textStyle,
}) => {
  return (
    <TouchableOpacity
      style={[
        styles.base,
        styles[variant],
        styles[size],
        disabled && styles.disabled,
        style,
      ]}
      onPress={onPress}
      disabled={disabled || loading}
    >
      {loading ? (
        <ActivityIndicator
          color={variant === 'outline' ? theme.COLORS.primary.green : theme.COLORS.ui.background}
          size="small"
        />
      ) : (
        <Typography
          variant="button"
          style={[
            styles.text,
            styles[`${variant}Text` as keyof typeof styles] as TextStyle,
            disabled && styles.disabledText,
            textStyle,
          ].filter(Boolean) as TextStyle}
        >
          {title}
        </Typography>
      )}
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  base: {
    borderRadius: theme.BORDER_RADIUS.lg,
    alignItems: 'center',
    justifyContent: 'center',
    flexDirection: 'row',
    shadowColor: theme.COLORS.ui.accent,
    shadowOffset: { width: 0, height: 2 },
    shadowOpacity: 0.3,
    shadowRadius: 8,
    elevation: 4,
  },
  primary: {
    backgroundColor: theme.COLORS.ui.accent,
    borderWidth: 0,
    shadowColor: theme.COLORS.ui.accent,
    shadowOpacity: 0.5,
    shadowRadius: 12,
  },
  secondary: {
    backgroundColor: 'rgba(65, 105, 225, 0.1)',
    borderWidth: 1,
    borderColor: theme.COLORS.ui.accent,
  },
  outline: {
    backgroundColor: 'transparent',
    borderWidth: 1,
    borderColor: theme.COLORS.ui.border,
  },
  small: {
    paddingVertical: theme.SPACING.sm,
    paddingHorizontal: theme.SPACING.lg,
    minWidth: 100,
  },
  medium: {
    paddingVertical: theme.SPACING.md,
    paddingHorizontal: theme.SPACING.xl,
    minWidth: 150,
  },
  large: {
    paddingVertical: theme.SPACING.lg,
    paddingHorizontal: theme.SPACING.xl * 1.5,
    minWidth: 200,
  },
  disabled: {
    backgroundColor: theme.COLORS.ui.disabled,
    borderColor: 'transparent',
    shadowOpacity: 0,
  },
  text: {
    textAlign: 'center',
  },
  primaryText: {
    color: theme.COLORS.ui.text,
    textShadowColor: 'rgba(0, 0, 0, 0.2)',
    textShadowOffset: { width: 0, height: 1 },
    textShadowRadius: 2,
  },
  secondaryText: {
    color: theme.COLORS.ui.accent,
    textShadowColor: theme.COLORS.ui.accent,
    textShadowOffset: { width: 0, height: 0 },
    textShadowRadius: 4,
  },
  outlineText: {
    color: theme.COLORS.ui.text,
  },
  disabledText: {
    color: theme.COLORS.ui.textSecondary,
  },
}); 