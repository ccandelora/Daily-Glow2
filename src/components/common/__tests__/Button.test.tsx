import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import { Button } from '../Button';
import { TouchableOpacity, StyleProp, ViewStyle } from 'react-native';
import theme from '@/constants/theme';

describe('Button Component', () => {
  test('renders correctly with default props', () => {
    const onPressMock = jest.fn();
    const { getByText } = render(<Button title="Test Button" onPress={onPressMock} />);
    
    const buttonText = getByText('Test Button');
    expect(buttonText).toBeTruthy();
  });
  
  test('calls onPress when pressed', () => {
    const onPressMock = jest.fn();
    const { getByText } = render(<Button title="Test Button" onPress={onPressMock} />);
    
    const buttonElement = getByText('Test Button');
    fireEvent.press(buttonElement);
    
    expect(onPressMock).toHaveBeenCalledTimes(1);
  });
  
  test('does not call onPress when disabled', () => {
    const onPressMock = jest.fn();
    const { getByText } = render(
      <Button title="Test Button" onPress={onPressMock} disabled={true} />
    );
    
    const buttonElement = getByText('Test Button');
    fireEvent.press(buttonElement);
    
    expect(onPressMock).not.toHaveBeenCalled();
  });
  
  test('does not call onPress when loading', () => {
    const onPressMock = jest.fn();
    const { queryByText, getByTestId } = render(
      <Button title="Test Button" onPress={onPressMock} loading={true} />
    );
    
    // When loading, it should show activity indicator instead of text
    expect(queryByText('Test Button')).toBeNull();
    expect(queryByText('ActivityIndicator')).toBeDefined();
  });
  
  test('renders with correct variant styles', () => {
    const onPressMock = jest.fn();
    
    // Use a simple approach with snapshots for styling
    const { toJSON, rerender } = render(
      <Button title="Primary Button" onPress={onPressMock} />
    );
    
    // Primary button snapshot
    expect(toJSON()).toMatchSnapshot('primary-button');
    
    // Secondary variant
    rerender(<Button title="Secondary Button" onPress={onPressMock} variant="secondary" />);
    expect(toJSON()).toMatchSnapshot('secondary-button');
    
    // Outline variant
    rerender(<Button title="Outline Button" onPress={onPressMock} variant="outline" />);
    expect(toJSON()).toMatchSnapshot('outline-button');
  });
  
  test('renders with correct size styles', () => {
    const onPressMock = jest.fn();
    
    // Use snapshots for size testing too
    const { toJSON, rerender } = render(
      <Button title="Small Button" onPress={onPressMock} size="small" />
    );
    
    // Small button snapshot
    expect(toJSON()).toMatchSnapshot('small-button');
    
    // Medium size (default)
    rerender(<Button title="Medium Button" onPress={onPressMock} />);
    expect(toJSON()).toMatchSnapshot('medium-button');
    
    // Large size
    rerender(<Button title="Large Button" onPress={onPressMock} size="large" />);
    expect(toJSON()).toMatchSnapshot('large-button');
    
    // Compact size
    rerender(<Button title="Compact Button" onPress={onPressMock} size="compact" />);
    expect(toJSON()).toMatchSnapshot('compact-button');
  });
}); 