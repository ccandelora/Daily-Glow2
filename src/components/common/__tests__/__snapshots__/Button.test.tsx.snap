// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component renders with correct size styles: compact-button 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#4169E1",
      "borderRadius": 16,
      "borderWidth": 0,
      "elevation": 4,
      "flexDirection": "row",
      "justifyContent": "center",
      "minWidth": 0,
      "opacity": 1,
      "paddingHorizontal": 16,
      "paddingVertical": 8,
      "shadowColor": "#4169E1",
      "shadowOffset": {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.5,
      "shadowRadius": 12,
    }
  }
>
  <Text
    adjustsFontSizeToFit={true}
    numberOfLines={1}
    style={
      [
        {
          "color": "#FFFFFF",
          "fontFamily": "System",
          "fontSize": 16,
        },
        {
          "fontSize": 16,
          "fontWeight": "600",
          "textAlign": "center",
        },
        undefined,
        null,
        [
          {
            "textAlign": "center",
          },
          {
            "color": "#FFFFFF",
            "textShadowColor": "rgba(0, 0, 0, 0.2)",
            "textShadowOffset": {
              "height": 1,
              "width": 0,
            },
            "textShadowRadius": 2,
          },
          {
            "fontSize": 14,
          },
        ],
      ]
    }
  >
    Compact Button
  </Text>
</View>
`;

exports[`Button Component renders with correct size styles: large-button 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#4169E1",
      "borderRadius": 16,
      "borderWidth": 0,
      "elevation": 4,
      "flexDirection": "row",
      "justifyContent": "center",
      "minWidth": 200,
      "opacity": 1,
      "paddingHorizontal": 48,
      "paddingVertical": 24,
      "shadowColor": "#4169E1",
      "shadowOffset": {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.5,
      "shadowRadius": 12,
    }
  }
>
  <Text
    adjustsFontSizeToFit={true}
    numberOfLines={1}
    style={
      [
        {
          "color": "#FFFFFF",
          "fontFamily": "System",
          "fontSize": 16,
        },
        {
          "fontSize": 16,
          "fontWeight": "600",
          "textAlign": "center",
        },
        undefined,
        null,
        [
          {
            "textAlign": "center",
          },
          {
            "color": "#FFFFFF",
            "textShadowColor": "rgba(0, 0, 0, 0.2)",
            "textShadowOffset": {
              "height": 1,
              "width": 0,
            },
            "textShadowRadius": 2,
          },
        ],
      ]
    }
  >
    Large Button
  </Text>
</View>
`;

exports[`Button Component renders with correct size styles: medium-button 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#4169E1",
      "borderRadius": 16,
      "borderWidth": 0,
      "elevation": 4,
      "flexDirection": "row",
      "justifyContent": "center",
      "minWidth": 150,
      "opacity": 1,
      "paddingHorizontal": 32,
      "paddingVertical": 16,
      "shadowColor": "#4169E1",
      "shadowOffset": {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.5,
      "shadowRadius": 12,
    }
  }
>
  <Text
    adjustsFontSizeToFit={true}
    numberOfLines={1}
    style={
      [
        {
          "color": "#FFFFFF",
          "fontFamily": "System",
          "fontSize": 16,
        },
        {
          "fontSize": 16,
          "fontWeight": "600",
          "textAlign": "center",
        },
        undefined,
        null,
        [
          {
            "textAlign": "center",
          },
          {
            "color": "#FFFFFF",
            "textShadowColor": "rgba(0, 0, 0, 0.2)",
            "textShadowOffset": {
              "height": 1,
              "width": 0,
            },
            "textShadowRadius": 2,
          },
        ],
      ]
    }
  >
    Medium Button
  </Text>
</View>
`;

exports[`Button Component renders with correct size styles: small-button 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#4169E1",
      "borderRadius": 16,
      "borderWidth": 0,
      "elevation": 4,
      "flexDirection": "row",
      "justifyContent": "center",
      "minWidth": 100,
      "opacity": 1,
      "paddingHorizontal": 24,
      "paddingVertical": 8,
      "shadowColor": "#4169E1",
      "shadowOffset": {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.5,
      "shadowRadius": 12,
    }
  }
>
  <Text
    adjustsFontSizeToFit={true}
    numberOfLines={1}
    style={
      [
        {
          "color": "#FFFFFF",
          "fontFamily": "System",
          "fontSize": 16,
        },
        {
          "fontSize": 16,
          "fontWeight": "600",
          "textAlign": "center",
        },
        undefined,
        null,
        [
          {
            "textAlign": "center",
          },
          {
            "color": "#FFFFFF",
            "textShadowColor": "rgba(0, 0, 0, 0.2)",
            "textShadowOffset": {
              "height": 1,
              "width": 0,
            },
            "textShadowRadius": 2,
          },
        ],
      ]
    }
  >
    Small Button
  </Text>
</View>
`;

exports[`Button Component renders with correct variant styles: outline-button 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "transparent",
      "borderColor": "rgba(82, 67, 194, 0.3)",
      "borderRadius": 16,
      "borderWidth": 1,
      "elevation": 4,
      "flexDirection": "row",
      "justifyContent": "center",
      "minWidth": 150,
      "opacity": 1,
      "paddingHorizontal": 32,
      "paddingVertical": 16,
      "shadowColor": "#4169E1",
      "shadowOffset": {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.3,
      "shadowRadius": 8,
    }
  }
>
  <Text
    adjustsFontSizeToFit={true}
    numberOfLines={1}
    style={
      [
        {
          "color": "#FFFFFF",
          "fontFamily": "System",
          "fontSize": 16,
        },
        {
          "fontSize": 16,
          "fontWeight": "600",
          "textAlign": "center",
        },
        undefined,
        null,
        [
          {
            "textAlign": "center",
          },
          {
            "color": "#FFFFFF",
          },
        ],
      ]
    }
  >
    Outline Button
  </Text>
</View>
`;

exports[`Button Component renders with correct variant styles: primary-button 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#4169E1",
      "borderRadius": 16,
      "borderWidth": 0,
      "elevation": 4,
      "flexDirection": "row",
      "justifyContent": "center",
      "minWidth": 150,
      "opacity": 1,
      "paddingHorizontal": 32,
      "paddingVertical": 16,
      "shadowColor": "#4169E1",
      "shadowOffset": {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.5,
      "shadowRadius": 12,
    }
  }
>
  <Text
    adjustsFontSizeToFit={true}
    numberOfLines={1}
    style={
      [
        {
          "color": "#FFFFFF",
          "fontFamily": "System",
          "fontSize": 16,
        },
        {
          "fontSize": 16,
          "fontWeight": "600",
          "textAlign": "center",
        },
        undefined,
        null,
        [
          {
            "textAlign": "center",
          },
          {
            "color": "#FFFFFF",
            "textShadowColor": "rgba(0, 0, 0, 0.2)",
            "textShadowOffset": {
              "height": 1,
              "width": 0,
            },
            "textShadowRadius": 2,
          },
        ],
      ]
    }
  >
    Primary Button
  </Text>
</View>
`;

exports[`Button Component renders with correct variant styles: secondary-button 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "rgba(65, 105, 225, 0.1)",
      "borderColor": "#4169E1",
      "borderRadius": 16,
      "borderWidth": 1,
      "elevation": 4,
      "flexDirection": "row",
      "justifyContent": "center",
      "minWidth": 150,
      "opacity": 1,
      "paddingHorizontal": 32,
      "paddingVertical": 16,
      "shadowColor": "#4169E1",
      "shadowOffset": {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.3,
      "shadowRadius": 8,
    }
  }
>
  <Text
    adjustsFontSizeToFit={true}
    numberOfLines={1}
    style={
      [
        {
          "color": "#FFFFFF",
          "fontFamily": "System",
          "fontSize": 16,
        },
        {
          "fontSize": 16,
          "fontWeight": "600",
          "textAlign": "center",
        },
        undefined,
        null,
        [
          {
            "textAlign": "center",
          },
          {
            "color": "#4169E1",
            "textShadowColor": "#4169E1",
            "textShadowOffset": {
              "height": 0,
              "width": 0,
            },
            "textShadowRadius": 4,
          },
        ],
      ]
    }
  >
    Secondary Button
  </Text>
</View>
`;
